modules/build/src/main/scala/scala/build/Build.scala

```diff
@@ -63,18 +62,17 @@ object Build {
       def foundMainClass =
         if (foundMainClasses0.isEmpty) Left(new NoMainClassFoundError)
         else if (foundMainClasses0.length == 1) Right(foundMainClasses0.head)
-        else {
-          val msg = "Found several main classes. Which would you like to run?"
-          val fallbackError = new SeveralMainClassesFoundError(
-            ::(foundMainClasses0.head, foundMainClasses0.tail.toList),
-            Nil
-          )
-          Interactive(options).chooseOne(
-            msg,
-            foundMainClasses0.toList,
-            fallbackError
-          )
-        }
+        else
+          options.Interactive.chooseOne(
+            "Found several main classes. Which would you like to run?",
+            foundMainClasses0.toList
+          ) match {
+            case Some(mainClass) => Right(mainClass)
+            case None => Left(new SeveralMainClassesFoundError(
+                ::(foundMainClasses0.head, foundMainClasses0.tail.toList),
+                Nil
+              ))
+          }
```

Comment:
`toRight` should allow to simplify this I think:
```suggestion
          ).toRight {
            new SeveralMainClassesFoundError(
              ::(foundMainClasses0.head, foundMainClasses0.tail.toList),
              Nil
            )
          }
```